plugins/modules/network/icx/icx_firmware_upgrade.py

```diff
@@ -0,0 +1,243 @@
+#!/usr/bin/python
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = """
+---
+module: icx_firmware_upgrade
+version_added: "1.2.0"
+author: "Ruckus Wireless (@Commscope)"
+short_description: Upgrades firmware of ICX switches
+description:
+  - This module copies new firmware to switch, flashes it to primary/secondary and reboots the switch.
+notes:
+  - Tested against ICX 10.1.
+  - For information on using ICX platform, see L(the ICX OS Platform Options guide,../network/user_guide/platform_icx.html).
+options:
+  server_type:
+    description:
+      - Data transfer protocol to be used
```

Comment:
```suggestion
      - Data transfer protocol to be used.
```
Every paragraph in a description needs to be a complete sentence (or multiple sentences).